packages/tao/src/shared/package-json.ts

```diff
@@ -0,0 +1,35 @@
+import { TargetConfiguration } from './workspace';
+
+export type PackageJsonTargetConfiguration = Omit<
+  TargetConfiguration,
+  'executor' | 'dependsOn'
+>;
+
+export interface NxProjectPackageJsonConfiguration {
+  targets: Record<keyof PackageJson['scripts'], PackageJsonTargetConfiguration>;
+}
+
+export interface PackageJson {
```

Comment:
I know `name`, `dependencies`, `devDependencies`, .etc are on `package.json` but that doesn't matter for this function so it's unnecessary.

Reply:
Yeah, I thought it may be good to have a more complete typing for `package.json` somewhere in the repo... I know there are at least a few incomplete ones spread throughout as inline types.

Do you think they should be removed?